contract/src/Custody.sol

```diff
@@ -27,459 +28,276 @@ contract Custody is IChannel, IDeposit {
-        }
-
-        uint256 participantsLength = meta.chan.participants.length;
-        for (uint256 i = 0; i < participantsLength; i++) {
-            address participant = meta.chan.participants[i];
-            _ledgers[participant].channels.remove(channelId);
-        }
-
-        // Mark channel as closed by removing it
-        delete _channels[channelId];
-
-        emit ChannelClosed(channelId);
-    }
-
-    /**
-     * @notice Internal function to ensure a channel exists
-     * @param channelId The channel identifier
-     * @return meta The channel's metadata
-     */
-    function _requireValidChannel(bytes32 channelId) internal view returns (Metadata storage meta) {
-        meta = _channels[channelId];
-        require(meta.chan.adjudicator != address(0), ChannelNotFound(channelId));
-        return meta;
-    }
-
-    /**
-     * @notice Internal function to validate a candidate state and handle the result
-     * @param meta The channel's metadata
-     * @param candidate The state to be adjudicated
-     * @param proofs Additional proof states if required
-     * @return valid True if the state is valid according to the adjudicator
-     */
-    function _validateState(Metadata storage meta, State memory candidate, State[] memory proofs)
-        internal
-        returns (bool valid)
-    {
-        IAdjudicator adjudicator = IAdjudicator(meta.chan.adjudicator);
-
-        // Adjudicators only validate state transitions, not channel status
-        try adjudicator.adjudicate(meta.chan, candidate, proofs) returns (bool result) {
-            valid = result;
-
-            if (valid) {
-                // Store the valid state
-                meta.lastValidState = candidate;
-
-                // Channel status handling is managed by Custody contract
-                // Don't automatically change status to ACTIVE here
-            } else {
-                // If adjudicator returns false, mark as invalid
-                meta.status = Status.INVALID;
-            }
-
-            return valid;
-        } catch {
-            // If the adjudicator call reverts, treat as invalid state
-            meta.status = Status.INVALID;
-            return false;
-        }
+      // Distribute funds according to allocations
+      uint256 allocsLength = meta.lastValidState.allocations.length;
+      for (uint256 i = 0; i < allocsLength; i++) {
+        Allocation memory allocation = meta.lastValidState.allocations[i];
+        _unlockChannelFundsToAccount(channelId, allocation.destination, allocation.token, allocation.amount);
+      }
+
+      uint256 participantsLength = meta.chan.participants.length;
+      for (uint256 i = 0; i < participantsLength; i++) {
+        address participant = meta.chan.participants[i];
+        _ledgers[participant].channels.remove(channelId);
+      }
+
+      // Mark channel as closed by removing it
+      delete _channels[channelId];
+
+      emit ChannelClosed(channelId);
```

Comment:
This should be moved 1 level up. It is a bad side-effect to this method.

Reply:
Ok made the edut

Reply:
1fcd517fd0752ce0986130383b1cda29faedf515